lib/pbench/server/api/resources/__init__.py

```diff
@@ -674,18 +674,43 @@ class API_OPERATION(Enum):
 
 
 class API_AUTHORIZATION(Enum):
+    """
+    Defines the mechanism by which ApiBase infrastructure will automatically
+    authorize the client for the API method:
+
+        NONE:           No authorization is necessary, or the API has special
+                        requirements and will authorize for itself
+        DATASET:        The client is authorized against the owner and access
+                        of the referenced dataset.
+        USER_ACCESS:    The client is authorized against the USER and ACCESS
+                        type parameters, which must each appear only once in
+                        the various schema for the HTTP method used.
+    """
+
     NONE = auto()
     DATASET = auto()
     USER_ACCESS = auto()
 
 
 class ApiParams(NamedTuple):
-    body: Optional[JSONOBJECT]
-    query: Optional[JSONOBJECT]
-    uri: Optional[JSONOBJECT]
+    """
+    Collect the JSON description of parameters to an API provides via the three
+    defined sources: the Flask URI template parameters, HTTP query parameters,
+    and the JSON request body.
```

Comment:
I'm having trouble parsing this text; remove the word "to" (or replace it with "which")?  (Or, remove the word "provides"?)